remote/save.go

```diff
@@ -16,72 +15,50 @@ func (i *Image) Save(additionalNames ...string) error {
 	return i.SaveAs(i.Name(), additionalNames...)
 }
 
-func (i *Image) SaveAs(name string, additionalNames ...string) error {
-	var err error
-	allNames := append([]string{name}, additionalNames...)
-
-	// create time
-	if i.image, err = mutate.CreatedAt(i.image, v1.Time{Time: i.createdAt}); err != nil {
-		return fmt.Errorf("setting creation time: %w", err)
-	}
-
-	// history
-	if i.image, err = imgutil.OverrideHistoryIfNeeded(i.image); err != nil {
-		return fmt.Errorf("overriding history: %w", err)
-	}
-	cfg, err := i.image.ConfigFile()
-	if err != nil {
-		return fmt.Errorf("getting config file: %w", err)
-	}
-	created := v1.Time{Time: i.createdAt}
-	if i.withHistory {
-		// set created
-		for j := range cfg.History {
-			cfg.History[j].Created = created
-		}
-	} else {
-		// zero history, set created
-		for j := range cfg.History {
-			cfg.History[j] = v1.History{Created: created}
-		}
-	}
-
-	// docker, container
-	cfg.DockerVersion = ""
-	cfg.Container = ""
+var (
+	emptyLayer   = static.NewLayer([]byte{}, types.OCILayer)
+	emptyHistory = v1.History{Created: v1.Time{Time: imgutil.NormalizedDateTime}}
+)
 
-	// commit config
-	i.image, err = mutate.ConfigFile(i.image, cfg)
-	if err != nil {
-		return fmt.Errorf("zeroing history: %w", err)
+func (i *Image) SaveAs(name string, additionalNames ...string) error {
+	if err := i.SetCreatedAtAndHistory(); err != nil {
+		return err
 	}
 
-	// layers
-	layers, err := i.image.Layers()
+	// add empty layer if needed
+	layers, err := i.Layers()
 	if err != nil {
 		return fmt.Errorf("getting layers: %w", err)
 	}
 	if len(layers) == 0 && i.addEmptyLayerOnSave {
-		empty := static.NewLayer([]byte{}, types.OCILayer)
-		i.image, err = mutate.AppendLayers(i.image, empty)
-		if err != nil {
+		if err = i.AddLayerWithHistory(emptyLayer, emptyHistory); err != nil {
 			return fmt.Errorf("adding empty layer: %w", err)
 		}
 	}
 
-	i.image, err = imgutil.MutateManifest(i.image, func(mfest *v1.Manifest) (mutateSubject, mutateAnnotations bool) {
+	i.Image, err = imgutil.MutateManifest(i.Image, func(mfest *v1.Manifest) (mutateSubject, mutateAnnotations bool) {
 		i.mutex.TryLock()
 		defer i.mutex.Unlock()
+
+		os, _ := i.OS()
+		arch, _ := i.Architecture()
+		variant, _ := i.Variant()
+		osv, _ := i.OSVersion()
+		features, _ := i.Features()
+		osFeatures, _ := i.OSFeatures()
+		urls, _ := i.URLs()
+		annotations, _ := i.Annotations()
+
```

Comment:
I believe this should be 
```go
                os, _ := i.os
		arch, _ := i.arch
		variant, _ := i.variant
		osv, _ := i.osVersion
		features, _ := i.features
		osFeatures, _ := i.osFeatures
		urls, _ := i.urls
		annotations, _ := i.annotations
```
as these fields will keep track of updated values, calling `i.OS()` might not return the value of `i.os`, 

- `i.OS()` will return the OS of the image that is before the user modified
- `i.os` will return the updated OS the user wants the image to be saved as.

for example, a user wants an Image with `linux/amd64` to be updated to `windows/amd64`
the `i.OS()` still returns `linux`, whereas `i.os` returns `windows`